### ISSUE TITLE
"`local_ctx` does not live long enough" compilation error on `rustc 1.64.0-nightly (62b272d25 2022-07-21)`

### ISSUE THREAD

**Issue by mitchmindtree** [rust-lang/mdBook#1860]
First noticed downstream in one of our CI passes [here](https://github.com/FuelLabs/sway/pull/2359#issuecomment-1192097008), it appears the latest nightly has trouble building `mdbook`.

This includes all versions at least from `0.4.19` through to the current HEAD of `master`.

```
error[E0597]: `local_ctx` does not live long enough
   --> src/renderer/html_handlebars/helpers/navigation.rs:154:25
    |
154 |             t.render(r, &local_ctx, &mut local_rc, out)
    |                         ^^^^^^^^^^ borrowed value does not live long enough
155 |         })?;
    |         -
    |         |
    |         `local_ctx` dropped here while still borrowed
    |         borrow might be used here, when `local_rc` is dropped and runs the destructor for type `handlebars::RenderContext<'_, '_>`
    |
    = note: values in a scope are dropped in the opposite order they are defined

For more information about this error, try `rustc --explain E0597`.
error: could not compile `mdbook` due to previous error
```

The issue appears to have been introduced in the latest nightly (`rustc 1.64.0-nightly (62b272d25 2022-07-21)`) as the previous nightly (`rustc 1.64.0-nightly (d68e7ebc3 2022-07-20)`) builds successfully.

Yet to dig in to what's going on here, but just thought I'd open an issue to track this first.

**Comment by mitchmindtree** [comment]
I've opened #1861 that appears to fix this.

I've also opened an issue upstream https://github.com/rust-lang/rust/issues/99591.

**Comment by andrewdavidmackenzie** [comment]
Just ran into this too, on linux (Ubuntu) only it seems

**Comment by ehuss** [comment]
Closing fixed by #1860.
